docker/scripts/p4runtime_switch.py

```diff
@@ -0,0 +1,321 @@
+# Copyright 2017-present Barefoot Networks, Inc.
+# Copyright 2017-present Open Networking Foundation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import sys, os, tempfile, socket
+from time import sleep
+import json
+
+from mininet.node import Switch
+from mininet.moduledeps import pathCheck
+from mininet.log import info, error, debug, setLogLevel
+
+#setLogLevel('debug')
+setLogLevel('info')
+
+import grpc
+from p4_mininet import P4Switch, SWITCH_START_TIMEOUT
+from netstat import check_listening_on_port
+import p4runtime_lib.bmv2
+import p4runtime_lib.helper
+from p4runtime_lib.error_utils import printGrpcError
+
+def tableEntryToString(flow):
+    if 'match' in flow:
+        match_str = ['%s=%s' % (match_name, str(flow['match'][match_name])) for match_name in
+                     flow['match']]
+        match_str = ', '.join(match_str)
+    elif 'default_action' in flow and flow['default_action']:
+        match_str = '(default action)'
+    else:
+        match_str = '(any)'
+    params = ['%s=%s' % (param_name, str(flow['action_params'][param_name])) for param_name in
+              flow['action_params']]
+    params = ', '.join(params)
+    return "%s: %s => %s(%s)" % (
+        flow['table'], match_str, flow['action_name'], params)
+
+# object hook for josn library, use str instead of unicode object
```

Comment:
are these the same methods as in simple_controller.py? maybe try not to duplicate them is possible.